### DISCUSSION TITLE
Working around sinon unable to stub ES modules

### DISCUSSION THREAD

**Discussion by kurtaking** [oclif/test#503] [Q&A]
I am trying to test a command that leverages `@inquirer/prompts`, and cannot figure out how to stub the input prompt.

Here is my test

```ts
import * as prompts from '@inquirer/prompts'
import {expect, test} from '@oclif/test'

import New from '../../../src/commands/config/new'

describe('config:new', () => {
  test
    .stdout({print: true})
    .stub(prompts, 'input', (stub) => stub.onFirstCall().resolves('First Last'))
    .do(() => New.run([]))
    .it('runs config:new', (ctx) => {
      expect(ctx.stdout).to.contain('Hello First!')
    })
})
```

I receive the following error on the `.stub` call.

```sh
 TypeError: ES Modules cannot be stubbed
```

Does anyone have a workaround?

**Comment by hahahakazi** [comment]
Here @inquirer/prompts is an ESM module, and sinon.stub or similar techniques won't work with ESM imports due to their immutability.  Because ESM imports are immutable
- If u want to use ESM module by using esmock (https://github.com/iambumblehead/esmock) you can mock the input method from @inquirer/prompts.
- Or you can change your project to CommonJS module for simplicity but the future flexibility might be limited. Here is the stackOverflow discussion (https://stackoverflow.com/questions/50763323/stub-an-export-from-a-native-es-module-without-babel).
